examples/src/main/java/org/apache/spark/examples/ml/JavaQuantileDiscretizerExample.java

```diff
@@ -54,14 +54,15 @@ public static void main(String[] args) {
     });
 
     Dataset<Row> df = spark.createDataFrame(data, schema);
-
-    // Note that we compute exact quantiles here by setting `relativeError` to 0 for
-    // illustrative purposes, however in most cases the default parameter value should suffice
+    // $example off$
+    // Output of QuantileDiscretizer for such small datasets differ wrt underlying cores.
+    // Allocating single partition for the dataframe helps with consistent results.
```

Comment:
Can me make this "Here we force a single partition to ensure consistent results. Note this is not necessary for normal use cases".

It must be clear not to do this in normal usage